libgssapi/src/util.rs

```diff
@@ -202,7 +203,11 @@ impl<'a> Deref for BufRef<'a> {
     type Target = [u8];
 
     fn deref(&self) -> &Self::Target {
-        unsafe { slice::from_raw_parts(self.0.value.cast(), self.0.length as usize) }
+        if self.0.value.is_null() && self.0.length == 0 {
+            &[]
```

Comment:
Saw this used in a different deref, I guess if there's a lifetime you can return a reference to an empty slice